File: 1281-subtract-the-product-and-sum-of-digits-of-an-integer/1281-subtract-the-product-and-sum-of-digits-of-an-integer.java
class Solution {
    public int subtractProductAndSum(int n) {
        String s = Integer.toString(n);
        int prod = 1;
        int sum = 0;
        for(int i = 0; i < s.length(); i++){
           int digit = s.charAt(i) - '0';

            prod = prod * digit;
            sum = sum + digit;
        }
        return (prod - sum);
    }
}

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/leethub-v4/bcilpkkbokcopmabingnndookdogmbna

File: 1281-subtract-the-product-and-sum-of-digits-of-an-integer/1281-subtract-the-product-and-sum-of-digits-of-an-integer.c
int subtractProductAndSum(int n) {
    int prod = 1;
    int sum =0;

    while(n > 0){
        int digit = n % 10 ;
        
        prod = prod * digit ;
        sum = sum + digit;

        n = n/10;
    }
    return prod-sum;
}

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/leethub-v4/bcilpkkbokcopmabingnndookdogmbna